homework/weather/exercise1.md

```diff
@@ -0,0 +1,68 @@
+# Homework 1 - The Weather Widget
+
+* Example code from lecture 1 can be used as a basis.
+* Make sure you understand what each piece that you do use is doing and pick
+  only the part that are necessary to fulfil these requirements.
+* Read through all the requirements carefully
+* Ask any questions in the course Slack workspace
+
+## Configure initial project (1 / 10 points)
```

Comment:
I _recommend_ changing both of the setup points to `2` and the test/app points to `3` for 
1) better aligning the amount of requirements vs the amount of points
2) giving some points if just the setup is completed

Here is the grading scheme I used for `mathemagician` homework 1:

```
* *index.html* 
  * Loads JS and CSS created by the specified build tool (/1)
  * JS file executed after *DOM* has loaded (/0.5)
  * *title* set on the document (/0.5)
* App functionality
  * Next expression displayed after a correct answer (/2)
  * Duration displayed after 3 successful rounds (/1)
  * Unit tests (/2)
* Build setup
  * Page reloads automatically upon changes to source code (/0.5)
  * Source maps configured and working (/0.5)
  * Babel outputs JS compatible with required browsers (/1)
  * Linter configured and passes. Rules not simply disabled. (/1)
```

Reply:
For future inspiration, here are the grading schemes for subsequent homework submissions:

Homework2:
```
* Instructions with author name (/1)
* Rounds input (/1)
* Start button with validation (/1)
* Game progresses as prescribed (/3)
* Multiplication (/1)
* Unit tests for multiplication (/1)
* Minimally 5 React components (/1)
* Prop types validated (/1)
```

Homework 3:
```
* Number input focused on start-up (/1)
* Subtraction and division operators (/2)
  * Added to game (/1)
  * Unit tested (/1)
* Game continues on incorrect answer (/1)
* Round history (/3)
  * Present during game (/1)
  * Present in game over screen with time spent (/1)
  * Colouring (/1)
* React components (/2)
  * Maximally 1 class-based component (/1)
  * Smoke tests for components (/1)
* Game progression logic test (/1)
```

Homework 4:
```
* Skipping (/2)
* New game (/2)
* Game history including tests (/2)
* useReducer hook used (/2)
* 100% test coverage for reducers (/2)
```

Homework5:
```
* Create game POST requests sent (/2.5)
* Create creation in-flight and error cases handled (/1)
* Create guess POST requests sent (/2.5)
* Create guess in-flight and error cases handled (/1)
* Component tests (1/1)
* Integration test (1/1)
* Dispatch middleware (/1)
```

Homework 6:
```
* Connecting with player name (/2p)
* Views (connect button, connecting..., disconnect button) shown based on connection status (/1p)
* Disconnecting (/1p)
* Online players list shown (/2p)
* Connection error handling on code 4000 (/1p)
* Reducer de-composition (/1p)
* Reducers tested (/1p)
* Components tested (/1p)

Bonus:
 (you) shown based on player ID (/1p)
```

Homework 7:
```
* Welcome component shown when not connected (/2p)
* Links(5) at the top of the application navigate to respective page(/2p)
* /createGame screen (/1p)
* /players screen (/1p)
* /ongoingGames screen (/1p)
* /finishedGames screen (/1p)
* /games/:gameId screen (/2p)
```

Reply:
Thanks for the recommendations, it sounds good to me. Also, thanks for the grading schemes!